### DISCUSSION TITLE
bug: serializeTransaction function expect transaction.data but JSON RPC spec is transaction.input

### DISCUSSION THREAD

**Discussion by vincentlg** [wevm/viem#1898] [Idea / Feature Request]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

2.7.14

### Current Behavior

Get transactions with

```
 const block = await mainnetClient.getBlock({
        includeTransactions: true,
        blockNumber: 10593417,
      });
const transaction = block.transactions[0];

console.log(transaction);
/*
   {
     to: "0x...",
     value: "0x...",
     input: "0x....",
   }
*/
```
Then try to serialize a transaction with

`const value = serializeTransaction(transaction, transaction);`

The serializeTransaction function expects a transaction with a `data` field. But RPC returns a transaction with `input` instead. So, in `value`, the serialized transaction do not have any input.

To fix this issue, I have to overwrite the `data`field myself before calling serializeTransaction , like so:

```
if (!transaction.data) transaction.data = transaction.input;
```

Then, the result is as expected.

### Expected Behavior

I expect not to have to overwrite the `data` field.  `serializeTransaction` should directly be compatible with RPC result

### Link to Minimal Reproducible Example (StackBlitz, CodeSandbox, GitHub repo etc.)

https://stackblitz.com/edit/viem-getting-started-ksz1ya?file=index.ts

**Comment by jxom** [comment]
Moving this to feature request as it is not really a bug. 

A bit annoying that there is this type inconsistency in the JSON-RPC spec (ie. transaction requests have `data`, transaction responses have `input`). `serializeTransaction`'s main use-case is to serialize transaction requests, but we can look into tweaking the type if this is a real annoyance.
